liquidity-rewards/test/UniswapPoolRewardsTest.js

```diff
@@ -1,27 +1,276 @@
-const keccak256 = web3.utils.keccak256;
-const { expectEvent, expectRevert, time } = require('@openzeppelin/test-helpers');
+const { BN, time } = require('@openzeppelin/test-helpers');
+const { expect, assert } = require('chai');
 
+
+const ERC20Mock = artifacts.require('ERC20Mock');
 const UniswapPoolRewards = artifacts.require('UniswapPoolRewards');
 
-require('chai').should();
+const _1e18 = new BN('10').pow(new BN('18'));
+const AMONTH = new BN('2629800'); // 1 month = 2629800 seconds
+
+async function assertRevert(txPromise, message = undefined) {
+    try {
+      const tx = await txPromise
+      // console.log("tx succeeded")
+      assert.isFalse(tx.receipt.status) // when this assert fails, the expected revert didn't occur, i.e. the tx succeeded
+    } catch (err) {
+      // console.log("tx failed")
+      assert.include(err.message, "revert")
+      // TODO !!!
+      
+      // if (message) {
+      //   assert.include(err.message, message)
+      // }
+    }
+}
+
+const almostEqualDiv1e18 = function (expectedOrig, actualOrig) {
+  const expected = expectedOrig.div(_1e18);
+  const actual = actualOrig.div(_1e18);
+  this.assert(
+    expected.eq(actual) ||
+      expected.addn(1).eq(actual) || expected.addn(2).eq(actual) ||
+      actual.addn(1).eq(expected) || actual.addn(2).eq(expected),
+    'expected #{act} to be almost equal #{exp}',
+    'expected #{act} to be different from #{exp}',
+    expectedOrig.toString(),
+    actualOrig.toString(),
+  );
+};
+
+require('chai').use(function (chai, utils) {
+  chai.Assertion.overwriteMethod('almostEqualDiv1e18', function (original) {
+    return function (value) {
+      if (utils.flag(this, 'bignumber')) {
+        var expected = new BN(value);
+        var actual = new BN(this._obj);
+        almostEqualDiv1e18.apply(this, [expected, actual]);
+      } else {
+        original.apply(this, arguments);
+      }
+    };
+  });
+});
+
+contract('UniswapPoolRewards', function ([_, wallet1, wallet2, wallet3, wallet4, owner]) {
+  const deploy = async (that) => {
+      that.uni = await ERC20Mock.new('Uniswap token', 'LPT', owner, 0);
```

Comment:
I suggest renaming to match the actual token at https://etherscan.io/token/0xa0d230dca71a813c68c278ef45a7dac0e584ee61

```suggestion
      that.uni = await ERC20Mock.new('Uniswap V2', 'UNI-V2', owner, 0);
```